source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1187,6 +1305,29 @@ void Filter::mergePerRouteConfig() {
   }
 }
 
+void DeferredDeletableStream::closeStreamOnTimer(Filter* filter) {
+  // Close the stream.
+  if (stream_) {
+    ENVOY_LOG(debug, "Closing the stream");
+    if (stream_->close()) {
+      stats.streams_closed_.inc();
+    }
+    stream_.reset();
+  } else {
+    ENVOY_LOG(debug, "Stream already closed");
+  }
+
+  // Erase this entry from the map.
+  parent.erase(filter);
+}
+
+// In the deferred closure mode, stream closure is deferred upon filter destruction, with a timer
+// to prevent unbounded resource usage growth.
+void DeferredDeletableStream::deferredClose(Envoy::Event::Dispatcher& dispatcher, Filter* filter) {
+  derferred_close_timer = dispatcher.createTimer([this, filter] { closeStreamOnTimer(filter); });
+  derferred_close_timer->enableTimer(std::chrono::milliseconds(DEFAULT_CLOSE_TIMEOUT_MS));
```

Comment:
I think this needs to be configurable. We can reuse the drain timeout for H/2 connections in the cluster for Envoy gRPC. For Google gRPC this may need to be an additional configuration option.

Reply:
Yes, you are right. I actually have already left a TODO (at source/extensions/filters/http/ext_proc/ext_proc.h line 149) for this.

I planned to address this TODO right after this PR (which is major part of ob11y mode) is LGed/merged.